diwan/src/commands/edit_commands.rs

```diff
@@ -0,0 +1,20 @@
+pub enum TextAction {
+    Insert,
+    Delete,
+    Replace,
+}
+
+pub struct EditCommand {
+    action: TextAction,
+    // Command-related fields
+}
+
+impl EditCommand {
+    pub fn handle_insert(&self, text: &str) {
+        // Handle text insert command
+    }
+
+    pub fn handle_delete(&self) {
+        // Handle text delete command
+    }
+}
```

Comment:
```suggestion
    pub fn handle_replace(&self) {
        // Handle text replace command
    }
}
```
missing `handle_replace` method here, consider adding it 😄